src/middlewared/middlewared/plugins/jail.py

```diff
@@ -42,8 +44,27 @@ def query(self, filters=None, options=None):
         try:
             jail_dicts = ioc.IOCage(
                 jail=jail_identifier).get('all', recursive=recursive)
-            jails = [list(jail.values())[0] for jail in jail_dicts
-                     ] if jail_identifier is None else [jail_dicts]
+            for jail in jail_dicts:
+                jail = list(jail.values())[0]
+                if jail['dhcp'] == 'on':
+                    interface = jail["interfaces"].split(',')[0].split(
+                        ':')[0]
+                    uuid = jail['host_hostuuid']
+
+                    if interface == "vnet0":
+                        # Inside jails they are epairNb
+                        interface = \
+                            f"{interface.replace('vnet', 'epair')}b"
```

Comment:
That replace seems you of place, if interface == "vnet0" result will always be "epair0b"

Reply:
That's expected, every jail inside will be called epair0b with a default configuration, if the user specifically uses a different interface then it will not be.

Reply:
The comment says epairNb, perhaps interface = 'epair0b' would be more straightfoward